From ee8dbe35a9cca36f44e88c3ed723814177450f2f Mon Sep 17 00:00:00 2001
From: Jason Hu <jason.hu@intel.com>
Date: Wed, 10 Apr 2013 18:20:41 -0400
Subject: Add two bits in OMX buffer flag to indicate OMX buffer contain top and bottom field and correct field display ordering.

BZ: 99408

Category: feature differentiation
Domain: Video.Media-VPP
Origin:  internal
Upstream-Candidate: No

Change-Id: I3effa1f887c5fd68e644d27afbd5bc5db977cda8
Orig-Change-Id: Ia0a14e8cddcd28ee0fb95eb236cb6605bba45fef
Signed-off-by: Jason Hu <jason.hu@intel.com>
Reviewed-on: http://android.intel.com:8080/100951
Reviewed-by: Sidorov, Sergey <sergey.sidorov@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Wang, Lili A <lili.a.wang@intel.com>
Reviewed-by: Shi, PingX <pingx.shi@intel.com>
Tested-by: Shi, PingX <pingx.shi@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 include/media/openmax/OMX_Core.h |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/include/media/openmax/OMX_Core.h b/include/media/openmax/OMX_Core.h
index 9fb0f6f..75f4f2d 100644
--- a/include/media/openmax/OMX_Core.h
+++ b/include/media/openmax/OMX_Core.h
@@ -409,7 +409,17 @@ typedef struct OMX_PARAM_COMPONENTROLETYPE {
  */
 #define OMX_BUFFERFLAG_CODECCONFIG 0x00000080
 
+/* interlaced frame flag: This flag is set to indicate the buffer contains a 
+ * top and bottom field and display ordering is top field first.
+ * @ingroup buf
+ */
+#define OMX_BUFFERFLAG_TFF 0x00010000
 
+/* interlaced frame flag: This flag is set to indicate the buffer contains a 
+ * top and bottom field and display ordering is bottom field first.
+ * @ingroup buf
+ */
+#define OMX_BUFFERFLAG_BFF 0x00020000
 
 /** @ingroup buf */
 typedef struct OMX_BUFFERHEADERTYPE
-- 
1.7.4.1

